lib/devise/failure_app.rb

```diff
@@ -252,5 +246,17 @@ def is_flashing_format?
     def request_format
       @request_format ||= request.format.try(:ref)
     end
+
+    def relative_url_root
+      @relative_url_root ||= begin
+        config = Rails.application.config
+
+        config.try(:relative_url_root) || config.try(:action_controller).try(:relative_url_root)
```

Comment:
You are right, there is no need to use `try`, i will change it.
[guide#4.2.x](http://guides.rubyonrails.org/v4.2/configuring.html#configuring-action-controller)
[guide#4.1.x](http://guides.rubyonrails.org/v4.1/configuring.html#configuring-action-controller)
[guide#4.0.x](http://guides.rubyonrails.org/v4.0.8/configuring.html#configuring-action-controller)
[guide#3.2.x](http://guides.rubyonrails.org/v3.2.18/configuring.html#configuring-action-controller)
[guide#2.3.x](http://guides.rubyonrails.org/v2.3.11/configuring.html#configuring-action-controller)